### PR TITLE
feat: add database config

### PR DESCRIPTION
**Pull request by mattsse**
add config type for customizing database settings

@onbjerg can I add this here:

https://github.com/paradigmxyz/reth/blob/927a9965f41a35679ef349c21ecda362b152ff7d/crates/config/src/config.rs#L12-L23 
which would require a dependency to db crate

currently we're using this to open a DB

https://github.com/paradigmxyz/reth/blob/927a9965f41a35679ef349c21ecda362b152ff7d/crates/storage/db/src/implementation/mdbx/mod.rs#L59-L62

can I add another function that accepts the config as argument?

blocked by #3450

### CODE REVIEW

File: crates/storage/db/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/// Configuration the database
+///
+/// This allows for the configuration of the database via [Environment](reth_libmdbx::Environment)
```
[the remark this answers is not in the document]
Comment (mattsse) [reply crates/storage/db/src/config.rs:3]: what's the suggestion here? less detailed docs, remove refs to mdbx things?

[the remark this answers is not in the document]
Comment (joshieDo) [reply crates/storage/db/src/config.rs:3]: remove refs to mdbx on the struct itself, but keep it on the fields that matter

**Comment by gakonst** [comment]
Maybe we should also allow the DB geometry to be exposed in reth.toml? If e.g. it makes syncing on certain devices easier? LGTM either way

**Comment by mattsse** [comment]
will integrate separately after #3450

**Comment by onbjerg** [comment]
> @onbjerg can I add this here:

yeah

> Maybe we should also allow the DB geometry to be exposed in reth.toml?

I don't think tweaking these settings per-device makes a lot of sense, afaik it is more workload dependent as opposed to platform dependent

@mattsse Please open up a follow up issue to add this to the book

**Comment by mattsse** [comment]
yeah now that we bumped the max readers I don't think we have a use case for this atm.

so I'm closing and we can revisit if that changes

**Pull request closed without merging**